flow/connectors/s3/s3.go

```diff
@@ -141,8 +141,8 @@ func ValidCheck(s3Client *s3.S3, bucketURL string, metadataDB *metadataStore.Pos
 }
 
 func (c *S3Connector) ConnectionActive() bool {
-	_, err := c.client.ListBuckets(nil)
-	return err == nil
+	_, listErr := c.client.ListBuckets(nil)
```

Comment:
This doesn't seem like a strong enough check for the connection being active.
Something along the lines of the `ValidCheck` function seems better, the bucket seems to be constant anyway.

Reply:
The ValidCheck function is performed as part of validation